cpp/include/utils/data_conversion.hpp

```diff
@@ -0,0 +1,58 @@
+/*
+    Data Conversion
+    -------------------
+    Upper/Lower case algorithm using bit manipulation.
+
+    Time complexity
+    ---------------
+    O(N), where N is the length of the input.
+
+    Space complexity
+    ----------------
+    O(N), where N is the result.
+ */
+
+#ifndef DATA_CONVERSION_HPP
+#define DATA_CONVERSION_HPP
+
+#include <string>
+
+#define UPPER_CASE_BIT_MASK 0b11011111
+#define LOWER_CASE_BIT_MASK 0b00100000
+
+bool is_letter(const char& c) {
```

Comment:
char& by ref 😄, auto generated by Jetbrain Clion because the `&` being used in `for` loop line 30 & 45

Reply:
Well spotted! All of these `char&` variables can be converted to `char` because creating references creates pointers, which would add unnecessary overhead for `char` values.

Reply:
Removed the ref from all char and all tests passes in my machine.